examples/tests/valid/api_cors_custom.test.w

```diff
@@ -25,16 +25,16 @@ test "GET /users has cors headers" {
   let response = http.get(api.url + "/users");
 
   let headers = response.headers;
-  t.Assert.equalNum(response.status, 200);
+  testing.Assert.equal(response.status, 200);
```

Comment:
Oooof, this is so cumbersome `testing.Assert.xxx()`...

I would love to find a good name for this module so we can avoid the class. I understand @Chriscbr that you are not comfortable with special casing so we can use `assert.equals()`, and I understand...

Some ideas:

```js
assertions.equal(x, y);
check.equal(x, y);
testing.equals(x, y);
asserts.equal(x, y);
assrt.equals(x, y);
ass.equals(x, y); // couldn't resist
tst.equals(x, y);
```

Any other ideas?

Reply:
> ass.equals(x, y); // couldn't resist

:laughing:

I would use `equal` over`equals`, and avoid the third person form in general.

Additionally, will there be a universal `not` for all assertions?

```js
assert.not.equal(x, y);
assert.notEqual(x, y);
!assert.equal(x, y);
```

Is it feasible to adopt Vitest's syntax?

```js
expect(x).toEqual(y);
expect(x).toStrictEqual(y);
expect(x).not.toEqual(y);
```

Reply:
> Is it feasible to adopt Vitest's syntax?

To some extent I think yes, but it wouldn't be possible to have this typed due to no support of generics in wing. 

Haven't tried the "not" part yet.

Having said that, I think the general sentiment was that "assert" style is preferred to "expect". 

It would solve the naming conflicts though 😀

Reply:
How about `expect` as the name of the module?

```js
bring expect;

expect.equals(x, y);
```

Reply:
> bring expect;
>
> expect.equals(x, y);

I'd sort of expect `expect` to behave similar to vitest / jest expectations, e.g. `expect(1).toEqual(1)`. 

I'm still wondering if this entire discussion would be obsolete if there was either of the following

1. A way to alias this https://github.com/winglang/wing/issues/4032 e.g. `bring "testing" as t ; t.assert.equals` or `bring "assertions" as foo`
1. Move the `assert` keyword into the `assert` module and make it something like `assert.ok` (pretty much exactly how [node is doing it](https://nodejs.org/api/assert.html), except for the [assert](https://nodejs.org/api/assert.html#assertvalue-message) alias)
1. Rename the bultin `assert` to e.g. `assert!`

Reply:
Not sure how #1 will solve this because users will still need to decide on an alternative name.

#2 hurts the ergonomics of the common case which is against our design principles.

#3 is interesting. I am wondering if maybe all of our builtin functions should use `!`. @Chriscbr @yoav-steinberg @MarkMcCulloh what do you think?

Reply:
I'd avoid adding special characters to our builtins since they aren't really special in any way and I don't think there's much value in flagging them as builtins. For all the user cares they're like any other function.

Reply:
> hurts the ergonomics of the common case which is against our design principles.

I disagree that a plain binary assert is a common case. I mean it's common in the sense that it's often available, but IMO it's not what people want most of the time. My guess is that asserting equality is the most common. I think having a simplistic global assert encourages crappily written assertions and may even hide rich assertion capabilities. Looking at the current assertions we have in the examples folder, *most of them are checking equality

<img width="493" alt="image" src="https://github.com/winglang/wing/assets/1237390/205a1384-1584-42e4-bc62-bd454ffe04ce">


> I am wondering if maybe all of our builtin functions should use !

I don't think we have (or should have) enough global functions to make this a useful convention. I don't have a particularly strong opinion on this I guess.
It's worth noting that `!` is not a valid identifier in js (or wing currently), so that could rear its head as an issue in the implementation.

Reply:
> #2 hurts the ergonomics of the common case which is against our design principles.

Is there a common case beyond testing? Haven't found reasoning about it, only [this](https://github.com/winglang/wing/issues/105) and [this](https://github.com/winglang/wing/pull/1030). Also, wouldn't this apply to something like [fetch](https://www.winglang.io/docs/standard-library/http/api-reference#@winglang/sdk.http.Util.fetch) as well, since it could be consider a pretty common case as well?

Reply:
re: an earlier suggestion -  naming the module `asserts` or `check` sounds like a decent option to me (we haven't really established a convention that modules must be singular nouns):

```js
bring asserts;

asserts.check(fluxCapacitor.enabled(), "forgot to turn on flux capacitor"); // or `asserts.isTrue()` maybe
asserts.notEqual(fluxCapacitor.power, 0, "more power is needed");
asserts.throws(() => { ... });

// or
bring check;

check.true(fluxCapacitor.enabled(), "forgot to turn on flux capacitor");
check.notEqual(fluxCapacitor.power, 0, "more power is needed");
check.throws(() => { ... });
```

Reply:
> Is there a common case beyond testing? 

Yes, defensive programming - expressing preconditions in your code.

> naming the module `asserts` or `check` sounds like a decent option

If we go in that direction, I think `expect` is actually a better name as it is more commonly used for this use case.

@skorfmann I hear what you are saying about expecting that `expect` will have a different style because of jest/vitest, but I am not sure it's a huge blocker from trying this name.

---

Let's move forward with `expect` please, and we will see how it feels. We can still change it later.

Reply:
Done